File: frontend/src/services/prescriptionService.js
import api from "../api/axios";

export const getPrescriptions = async (patientId) => {

    const response = await api.get(
        `/patients/${patientId}/prescriptions`
    );

    return response.data;

};

export const createPrescription = async (
    patientId,
    prescription
) => {

    const response = await api.post(
        `/patients/${patientId}/prescriptions`,
        prescription
    );

    return response.data;

};

export const updatePrescription = async (
    prescriptionId,
    prescription
) => {

    const response = await api.put(
        `/prescriptions/${prescriptionId}`,
        prescription
    );

    return response.data;

};

export const deletePrescription = async (
    prescriptionId
) => {

    await api.delete(
        `/prescriptions/${prescriptionId}`
    );

};

File: backend/src/main/java/com/zealthy/backend/service/PrescriptionService.java
package com.zealthy.backend.service;

import com.zealthy.backend.dto.request.PrescriptionRequest;
import com.zealthy.backend.dto.response.PrescriptionResponse;

import java.util.List;

public interface PrescriptionService {

    List<PrescriptionResponse> getPrescriptionsByPatient(Long patientId);

    PrescriptionResponse createPrescription(Long patientId,
                                            PrescriptionRequest request);

    PrescriptionResponse updatePrescription(Long prescriptionId,
                                            PrescriptionRequest request);

    void deletePrescription(Long prescriptionId);

}